lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -76,7 +78,13 @@ protected void setup(Binder binder)
         }
 
         newMapBinder(binder, String.class, TrinoFileSystemCache.class);
-        if (config.getCacheType() == null) {
+        if (config.getCacheType() != null) {
```

Comment:
It seems strange that we have a `MapBinder` for the cache implementation, but still need to conditionally install the modules. There should be a way to combine this, so that there is only a single place we look at the cache type.